File: day_02.R
commands <- read.table("data/input_02.txt",col.names = c("direction","magnitude"))

commands %<>%
  mutate(forward = magnitude * (direction=="forward")) %>% 
  mutate(down = magnitude * (direction=="down")) %>% 
  mutate(up = magnitude * (direction=="up")) %>% 
  mutate(aim = cumsum(down-up)) 

with(commands, sum(forward) * sum(down-up) )
with(commands, sum(forward) * sum(forward*aim) )

File: day_02.py
# part 1
data = []
with open("data/input_02.txt") as f:
    for line in f:
        direction, magnitude = line.strip().split()
        data.append((direction, int(magnitude)))

depth, horizontal = 0, 0
for direction, magnitude in data:
    if direction=="down":
        depth += magnitude
    elif direction=="up":
        depth -= magnitude
    else:
        horizontal += magnitude
print(depth*horizontal)

# part 2
depth, horizontal, aim = 0, 0, 0
for direction, magnitude in data:
    if direction=="down":
        aim += magnitude
    elif direction=="up":
        aim -= magnitude 
    else:
        horizontal += magnitude
        depth += aim*magnitude
print(depth*horizontal)
